software/base/src/main/java/brooklyn/entity/java/JavaSoftwareProcessSshDriver.java

```diff
@@ -272,69 +273,24 @@ public String getJmxContext() {
     }
 
     /**
-     * Checks for the presence of Java 6 or 7 on the entity's location, installing if necessary.
-     * @return true if Java 6 or 7 was found on the machine or if it was installed correctly, otherwise false.
+     * Checks for the presence of Java on the entity's location, installing if necessary.
+     * @return true if the required version of Java was found on the machine or if it was installed correctly,
+     * otherwise false.
      */
-    protected boolean checkForAndInstallJava6or7() {
-        Optional<String> version = getCurrentJavaVersion();
-        if (version.isPresent() && (version.get().startsWith("1.8") || version.get().startsWith("1.7") || version.get().startsWith("1.6"))) {
-            log.debug("Java version {} already installed at {}@{}", new Object[]{version.get(), getEntity(), getLocation()});
-            return true;
-        } else {
-            // Let's hope not!
-            if (version.isPresent()) {
-                log.debug("Found old Java version {} on {}@{}. Going to install latest Java version.",
-                        new Object[]{version.get(), getEntity(), getLocation()});
-            }
-            return tryJavaInstall("latest", BashCommands.installJava7Or6OrFail()) == 0;
-        }
-    }
-
-    /**
-     * Checks for the presence of Java 7 or 8 on the entity's location, installing if necessary.
-     * @return true if Java 7 or 8 was found on the machine or if it was installed correctly, otherwise false.
-     */
-    protected boolean checkForAndInstallJava7or8() {
-        Optional<String> version = getCurrentJavaVersion();
-        if (version.isPresent() && (version.get().startsWith("1.8") || version.get().startsWith("1.7"))) {
-            log.debug("Java version {} already installed at {}@{}", new Object[]{version.get(), getEntity(), getLocation()});
-            return true;
-        } else {
-            // Let's hope not!
-            if (version.isPresent()) {
-                log.debug("Found old Java version {} on {}@{}. Going to install latest Java version.",
-                        new Object[]{version.get(), getEntity(), getLocation()});
+    protected boolean checkForAndInstallJava(String requiredVersion) {
+        List<String> requiredVersionParts = Splitter.on(".").splitToList(requiredVersion);
+        int requiredJavaMinor = Integer.valueOf(requiredVersionParts.get(1));
```

Comment:
1.7 is necessary and sufficient.  7 is a maybe-nice-to-have i'd say.

--A


On 22/05/2015 14:17, Andrea Turli wrote:
>
> In 
> software/base/src/main/java/brooklyn/entity/java/JavaSoftwareProcessSshDriver.java 
> <https://github.com/apache/incubator-brooklyn/pull/619#discussion_r30893498>:
>
> > -     * Checks for the presence of Java 7 or 8 on the entity's location, installing if necessary.
> > -     * @return true if Java 7 or 8 was found on the machine or if it was installed correctly, otherwise false.
> > -     */
> > -    protected boolean checkForAndInstallJava7or8() {
> > -        Optional<String> version = getCurrentJavaVersion();
> > -        if (version.isPresent() && (version.get().startsWith("1.8") || version.get().startsWith("1.7"))) {
> > -            log.debug("Java version {} already installed at {}@{}", new Object[]{version.get(), getEntity(), getLocation()});
> > -            return true;
> > -        } else {
> > -            // Let's hope not!
> > -            if (version.isPresent()) {
> > -                log.debug("Found old Java version {} on {}@{}. Going to install latest Java version.",
> > -                        new Object[]{version.get(), getEntity(), getLocation()});
> > +    protected boolean checkForAndInstallJava(String requiredVersion) {
> > +        List<String> requiredVersionParts = Splitter.on(".").splitToList(requiredVersion);
> > +        int requiredJavaMinor = Integer.valueOf(requiredVersionParts.get(1));
>
> better to support |1.7|, |7| or both?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/apache/incubator-brooklyn/pull/619/files#r30893498>.
>